packages/compass-telemetry/src/provider.ts

```diff
@@ -25,12 +28,20 @@ export function createTelemetryLocator() {
 
 export function useTelemetry(): TrackFunction {
   const context = React.useContext(TelemetryContext);
+  const logger = useLogger('COMPASS-TELEMETRY');
```

Comment:
Just to highlight again, not a blocker: we are unnecessarily creating instances of logger and telemetry in every place where telemetry is used instead of doing this once in the provider. It's not a blocker, but redundant and means more extra work that runtime needs to do for no good reason